discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -240,6 +239,79 @@ def detach_all_test_disks(self, node_name):
                 tempfile.gettempdir()), "File unexpectedly not in tmp, avoiding deletion to be on the safe side"
             os.remove(source_file)
 
+    def attach_interface(self, node_name, network_xml, target_interface="vnet3"):
+        """
+        Create network and Interface. New interface will be attached to a given node.
+        """
+        network = self.create_network(network_xml)
+        interface_mac = self.add_interface(node_name, network.bridgeName(), target_interface)
+        return network, interface_mac
+
+    def create_network(self, network_xml):
+        """
+        Create a network from a given xml and return libvirt.virNetwork object
+        """
+        logging.info(f"Creating new network: {network_xml}")
+        network = self.libvirt_connection.networkCreateXML(network_xml)
+        if network is None:
+            raise Exception(f"Failed to create network: {network_xml}")
+        active = network.isActive()
+        if active is not 1:
```

Comment:
You shouldn't use `is` to compare integers. It would work for numbers <=256, but it's bad practice.
Explanation - https://stackoverflow.com/questions/306313/is-operator-behaves-unexpectedly-with-integers

Reply:
fixed